apps/dcos_l4lb/src/dcos_l4lb_lashup_vip_listener.erl

```diff
@@ -120,11 +124,34 @@ process_vip(Key, BEs) ->
 
 -spec(categorize_backends([backend()]) -> [{family(), [backend()]}]).
 categorize_backends(BEs) ->
-    lists:foldl(fun ({_AgentIP, {IP, _Port, _Weight}}=BE, Acc) ->
+    lists:foldl(fun (BE, Acc) ->
+        case BE of
+            {_AgentIP, {IP, _Port, _Weight}} ->
+                BE0 = BE;
+            {AgentIP, {IP, Port}} ->
+                BE0 = {AgentIP, {IP, Port, 1}}
```

Comment:
Correct, this handles the case were we get a backend that isn't weighted.

1.  In the 2nd case AgentIP is used since we want it to be bound on the match for that case. For the 1st case using the `_` prefix denotes a match, but unused binding, there's a linter check that checks for unused bindings.
2. Yea the `_Weight` binding was added in #194 since that is what changed the format of the `backend` type.